libs/x-content/src/main/java/org/elasticsearch/xcontent/Text.java

```diff
@@ -36,31 +31,37 @@ public static Text[] convertFromStringArray(String[] strings) {
         return texts;
     }
 
-    private BytesReference bytes;
+    private ByteBuffer bytes;
```

Comment:
I think this is a good change, but also that it might be better if this gets done in a smaller, self-contained preliminary PR.
It will reduce the complexity of this one, can be merged quickly, and it would be helpful to pinpoint and/or reduce the blast radius in case of problems